psamm/commands/genedelete.py

```diff
@@ -0,0 +1,113 @@
+    """Find reactions requiring a spcified gene and delete them from model
+    """
+    @classmethod
+    def init_parser(cls, parser):
+        parser.add_argument(
+            '--gene', metavar='genes', action=FilePrefixAppendAction,
+            type=str, default=[], help='Delete Multiple Genes From Model')
+        parser.add_argument('--objective', help='Reaction flux to maximize')
+        super(GeneDeletionCommand, cls).init_parser(parser)
+
+    def run(self):
+        if self._args.objective is not None:
+            obj_reaction = self._args.objective
+        else:
+            obj_reaction = self._model.get_biomass_reaction()
+            if obj_reaction is None:
+                raise CommandError('The biomass reaction was not specified')
+        logger.info('Running single gene deletion...')
+        genes = set()
+        gene_assoc = {}
+        for reaction in self._model.parse_reactions():
+            assoc = None
+            if reaction.genes is None:
+                continue
+            elif isinstance(reaction.genes, string_types):
+                assoc = boolean.Expression(reaction.genes)
+            else:
+                variables = [boolean.Variable(g) for g in reaction.genes]
+                assoc = boolean.Expression(boolean.And(*variables))
+            genes.update(v.symbol for v in assoc.variables)
+            gene_assoc[reaction.id] = assoc
+
+        reactions = set(self._mm.reactions)
+        start_time = time.time()
+        testing_genes = set(self._args.gene)
+        deleted_reactions = set()
+
+        logger.info('Trying model without gene {}...'.format(
+                    testing_genes))
+
+        for reaction in reactions:
+            if reaction not in gene_assoc:
+                continue
+            assoc = gene_assoc[reaction]
+            found = False
+            for gene in testing_genes:
+                if boolean.Variable(gene) in assoc.variables:
+                    found = True
+                    break
+            if found:
+                new_assoc = assoc.substitute(
+                    lambda v: v if v.symbol not in testing_genes else False)
+                if new_assoc.has_value() and not new_assoc.value:
+                    logger.info('Deletion reaction {}...'.format(reaction))
+                    deleted_reactions.add(reaction)
+
+        print (deleted_reactions)
+        solver = self._get_solver()
+        prob = fluxanalysis.FluxBalanceProblem(self._mm, solver)
+
+        flux_var = prob.get_flux_var(reaction)
+        prob.prob.add_linear_constraints(flux_var == 0)
+        prob.maximize(obj_reaction)
+        prob.get_flux(obj_reaction)
+        wild = prob.get_flux(obj_reaction)
+
+        for reaction in deleted_reactions:
+            flux_var = prob.get_flux_var(reaction)
+            prob.prob.add_linear_constraints(flux_var == 0)
+            prob.maximize(obj_reaction)
+            prob.get_flux(obj_reaction)
```

Comment:
No need to to solve the maximization for every loop iteration. Move this one indentation level up so it is not inside the loop.